### DISCUSSION TITLE
Errors in installing step

### DISCUSSION THREAD

**Discussion by digna199** [USC-ACTLab/crazyswarm#374] [Q&A]
Hello everyone, I'm currently following the documentation to install crazyswarm in ubuntu 18.04. I've used the following commands in the "Physical Robots and Simulation tab":
$ export CSW_PYTHON=[python2]

$ sudo apt install git swig lib${CSW_PYTHON}-dev ${CSW_PYTHON}-numpy ${CSW_PYTHON}-yaml ${CSW_PYTHON}-matplotlib gcc-arm-embedded libpcl-dev libusb-1.0-0-dev sdcc ros-melodic-vrpn

$ git clone https://github.com/USC-ACTLab/crazyswarm.git

All is well up to this point, but things start going wrong in the next step when I try running the build script. 
$ cd crazyswarm
$ ./build.sh

I get the following: 
Crazyradio PA build
[python2] setup.py build_ext --inplace
/bin/sh: 1: [python2]: not found
Makefile:9: recipe for target 'swig' failed
make: *** [swig] Error 127

For some reason if I run the following command before I run the build script it seems to work fine: 
$ export CSW_PYTHON=python2

The difference is I removed the square brackets and it seems to have worked?

But then when I try moving on to the last step where I verify installation with the unit tests, it fails and says "28 failed in 1.37 seconds".

I've been grappling with the installation step for quite some while now trying whatever I can think of to get it to work but I keep running into errors. Could someone please advise on what might be going wrong? I appreciate all the help given in advance.

**Comment by jpreiss** [comment]
It is correct to remove the square brackets around python2. Which test is failing?

**Comment by sarahterracina** [comment]
I have a similar issue, the installation step 4  ./buildSimOnly.sh 
results in a 
/bin/sh: 1: [python2]: not found
Makefile:9: recipe for target 'swig' failed
make: *** [swig] Error 127

but when I go to install swig, it says I already have the latest version?

**Reply by jpreiss** [reply]
are you running the command

    export CSW_PYTHON=python2

or

    export CSW_PYTHON=[python2]

the first one is correct.

**Reply by Iko-Git** [reply]
I have just tried `CSW_PYTHON=python3` and got the following error while building the script via `$ ./buildSimOnly.sh` 
```
x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-
Bsymbolic-functions -Wl,-z,relro -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time 
-D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.6/../../../../../../crazyflie-firmware/src/modules
/src/collision_avoidance.o build/temp.linux-x86_64-3.6/../../../../../../crazyflie-firmware/src/modules/src/planner.o 
build/temp.linux-x86_64-3.6/../../../../../../crazyflie-firmware/src/modules/src/pptraj.o build/temp.linux-x86_64-3.6/../..
/../../../../crazyflie-firmware/src/modules/src/pptraj_compressed.o build/temp.linux-x86_64-3.6/cffirmware_wrap.o -o 
/home/user/crazyswarm/ros_ws/src/crazyswarm/scripts/pycrazyswarm/cfsim/_cffirmware.cpython-36m-x86_64-linux-
gnu.so
```

**Reply by Iko-Git** [reply]
got solved by running `export CSW_PYTHON=python2`
